radon/cli/tools.py

```diff
@@ -62,8 +63,12 @@ def _open(path):
         if path == '-':
             yield sys.stdin
         else:
-            with _open_function(path, encoding=_encoding) as f:
-                yield f
+            try:
+                with tokenize.open(path) as f:
+                    yield f
+            except:
```

Comment:
This bare except will eat an awful lot of errors...  Perhaps `elif hasattr(tokenize, 'open'):` would be better?

This still doesn't solve the problem on Python < 3.2 though..

Reply:
For Python < 3.2 we can just copy `tokenize.open` in Radon's code. The except was meant to capture encoding issues. There are a few of them and I didn't know where to find a complete list.

Reply:
Not sure which encoding (decoding?) issues you are thinking of, but the except here probably just hides them (and any exceptions coming from the consumer of the `yield f`).  I've created PR #131 which pulls in tokenize.open from CPython latest (by itself that was not enough).